### PR TITLE
OCM-2368 | feat: Makefile, sample code + tests, gitignore

### PR DESCRIPTION
**Pull request by hunterkepley**
JIRA: [OCM-2368](https://issues.redhat.com/browse/OCM-2368)

This story is about bootstrapping the repo, pretty much providing sample unit tests, an easy way to run those tests (`make test`), other useful things in the makefile, and basic project structure. This also includes the git commit msg check shell script that was in ROSA.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+.PHONY: fmt
+fmt:
+	gofmt -s -l -w cmd pkg
+
+.PHONY: lint
+lint:
+	golangci-lint run --timeout 5m0s
+
+.PHONY: commits/check
```
Comment (ciaranRoche) [inline Makefile:47]: Not needed, only in ROSA for Prow

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+commits/check:
+	@./hack/commit-msg-verify.sh
+
+.PHONY: clean
+clean:
+	rm -rf \
+		ocm-common \
+		$(NULL)
+
+.PHONY: generate
```
Comment (ciaranRoche) [inline Makefile:57]: Not needed right now, we can remove this

File: hack/commit-msg-verify.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
```
Comment (ciaranRoche) [inline hack/commit-msg-verify.sh:1]: Only needed for prow, we can remove this

File: Makefile (hunk 1 above)
Comment (gdbranco) [reply Makefile:47]: I'd say it is probably good having the commit checks though. Can help tracking it down and helps in ensuring people are not having some difficult to interpret messages

File: hack/commit-msg-verify.sh (hunk 3 above)
Comment (gdbranco) [reply hack/commit-msg-verify.sh:1]: We do have GH actions on this org right? We could call the script in the action :thinking:

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+generate:
+	which go-bindata || GO111MODULE=off go get -u github.com/go-bindata/go-bindata/...
+	go-bindata -nometadata -nocompress -pkg assets -o ./assets/bindata.go ./templates/...
+
+mocks:
+	mockgen --build_flags=--mod=mod -package mocks -destination=pkg/aws/mocks/iamapi.go github.com/aws/aws-sdk-go/service/iam/iamiface IAMAPI
```
Comment (gdbranco) [inline Makefile:63]: Probably don't need the mocks, at least not just yet

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
```
Comment (gdbranco) [inline Makefile:2]:
```suggestion
# Copyright (c) 2023 Red Hat, Inc.
```

File: hack/commit-msg-verify.sh (hunk 3 above)
Comment (hunterkepley) [reply hack/commit-msg-verify.sh]: @gdbranco https://github.com/openshift-online/ocm-common/pull/3 -- added in this MR already :)

File: pkg/sample/sample.go [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package sample
+
+// Add takes two integer inputs and returns the sum of the two inputs
+func Add(a int, b int) int {
```
Comment (gdbranco) [inline pkg/sample/sample.go:4]: Are we doing this just to have something? Can we include something more meaningful?

Comment (ciaranRoche) [reply pkg/sample/sample.go:4]: Just to have something right now, placeholders till we progress this epic.

**Pull request merged**